From b878ad1fa9dd0698158eddd55b4336034b53336d Mon Sep 17 00:00:00 2001
From: Grigory Vasilyev <h0tc0d3@gmail.com>
Date: Tue, 9 Jul 2024 11:23:45 +0300
Subject: [PATCH] winewayland.drv: Enable support wl_shm protocol version 2.

If the wayland compositor suports wl_shm v2 then use it.
---
 dlls/winewayland.drv/wayland.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/winewayland.drv/wayland.c b/dlls/winewayland.drv/wayland.c
index c94b35e173c..2ca8cc3bddd 100644
--- a/dlls/winewayland.drv/wayland.c
+++ b/dlls/winewayland.drv/wayland.c
@@ -127,7 +127,8 @@ static void registry_handle_global(void *data, struct wl_registry *registry,
     }
     else if (strcmp(interface, "wl_shm") == 0)
     {
-        process_wayland.wl_shm = wl_registry_bind(registry, id, &wl_shm_interface, 1);
+        process_wayland.wl_shm = wl_registry_bind(registry, id, &wl_shm_interface,
+                                                  version < 2 ? 1 : 2);
     }
     else if (strcmp(interface, "wl_seat") == 0)
     {
-- 
GitLab

